cfgtool

```diff
@@ -238,7 +259,103 @@ Stay safe. Use crypto. Don´t drink and factorize."
 
 }
 
+#Private function to easily add andzip the whole user temp dir and encrypt it with a password
+__add_and_zip_user() {
+  name=$1;
+  password=$2;
+  tmp_dir=$3;
+  client_name="vpn_${VPN_NAME}_${name}"
+  client_vpn_dirname="${tmp_dir}/${client_name}"
+  client_vpn_zipname="$vpn_${VPN_NAME}_${name}.zip"
+
+  ${script_file} add_user $name $tmp_dir
+  cd $tmp_dir
+  zip -P $password -r $client_vpn_zipname ./$client_name
+  cd ${working_dir}
+  er "
+The user's configuration has been created under ${client_vpn_zipname}.
+
+And encrypted with the generated password.
+
+Please transmit the zip over a safe channel to it's user.
+
+In order to access the VPN the user must unpack the tar, make sure the 'nobody' group and user exists on their system and execute 'openvpn openvpn-client.cfg' inside the unpacked directory as root.
+
+Stay safe. Use crypto. Don´t drink and factorize."
+
+}
+
+#http://blog.pragbits.com/it/2015/02/09/slack-notifications-via-curl/
+__notify_slack() {    # format message as a code block ```${msg}```
+    SLACK_TITLE=$1
+    SLACK_BODY=$2
+    SLACK_URL="https://hooks.slack.com/services/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"
+    curl -X POST --data "payload={
+     \"attachments\":[
+        {
+           \"color\":\"#00ff00\",
+           \"fields\":[
+              {
+                 \"title\":\"${SLACK_TITLE}\",
+                 \"value\":\"${SLACK_BODY}\",
+                 \"short\":false
+              }
+           ]
+        }
+     ]
+   }" ${SLACK_URL}
+ }
+#Load a textfile filled with the emails of the new users
+load_clients() {
+  #Filename of the textfile with clients
+  file_name="$1"; shift || fail "USAGE: ${script_file} load_clients <filename>"
+  #Read the text file line by line and add to array
+  IFS=$'\n' clients=($(<$file_name))
+  tmp_dir="./temp_clients"
+  date=`date +%Y-%m-%d:%H:%M:%S`
+  #Clean previous generated files
+  rm -rf "generated_clients.txt" $tmp_dir
+
+  #Rebuild the ca
+  yes YES | vpn_pki_setup
+
+  #generate new keys and zip them
+  for client in ${clients[@]}
+  do
+    cd ${working_dir}
+    name=$(echo $client | awk -F @ '{ print $1 }')
+    password=$(pwgen -s 256 -1 -v -c -0)
+    __add_and_zip_user $name $password $tmp_dir
+    zip_name="$vpn_${VPN_NAME}_${name}.zip"
+    test -f "${tmp_dir}/${zip_name}" ||
```

Comment:
How would this happen? Why not just check the return status of __add_and_zip_user